prov/cxi/src/cxip_msg_hpc.c

```diff
@@ -4196,12 +4197,12 @@ cxip_recv_common(struct cxip_rxc *rxc, void *buf, size_t len, void *desc,
 
 	assert(rxc_hpc->base.protocol == FI_PROTO_CXI);
 
+	ofi_genlock_lock(&rxc->ep_obj->lock);
 	ret = cxip_recv_req_init(rxc, buf, len, src_addr, tag, ignore, flags,
 				 tagged, context, comp_cntr, &req);
 	if (ret)
 		goto err;
```

Comment:
err case needs an unlock. double check the other error cases in this function - like line 4210 can be a ret=0 goto out (rename err to out and just have an unlock). or something like that. whatever you feel